### DISCUSSION TITLE
Why does serialize response always perform f

### DISCUSSION THREAD

**Discussion by ollz272** [fastapi/fastapi#9550] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get("/endpoint", response_model=SomePydanticModel)
def endpoint(self)
    return SomePydanticModel()
```

### Description

In this case, we have created a pydantic object within the endpoint and are returning it directly. However the method serialize_response seems to then:
- Turn this object into a dict
- Rerun all the pydantic validation
- Return the same object

For models with large amounts of data, this is quite expensive. Im sure there are many cases where its worth putting it through validation again, however for simple cases like this, is there a way we can check that the type of the response matches the expected type and, if it does, forgo fastapi redoing the validation?

### Operating System

macOS

### FastAPI Version

0.95.2

### Python Version

3.11.1

**Comment by karolpawlowski** [comment]
Bumping this one.

I recently observed that `fastapi.serialize_response` has very big impact - much bigger than querying database, when trying to return a list of many pydantic objects parsed into json.

FastAPI Version
0.111.0

Python Version
3.11.9

Pydantic version
1.10.13

**Reply by YuriiMotov** [reply, marked as answer]
Related question: https://github.com/tiangolo/fastapi/discussions/11694

**Comment by YuriiMotov** [comment]
Closing this as duplicating #11694
